src/Microsoft.VisualStudio.Editor.Razor/DefaultVisualStudioRazorParser.cs

```diff
@@ -388,7 +387,7 @@ internal void OnDocumentStructureChanged(object state)
         private void ConfigureTemplateEngine(IRazorEngineBuilder builder)
         {
             builder.Features.Add(new VisualStudioParserOptionsFeature(_documentTracker.EditorSettings));
-            builder.Features.Add(new VisualStudioTagHelperFeature(TextBuffer));
+            builder.Features.Add(new VisualStudioTagHelperFeature(_documentTracker.TagHelpers));
```

Comment:
❤️

Reply:
This seems wrong. Every time the TagHelpers are updated we don't plan on mutating that list do we? 

With this approach the feature will be stuck to 1 set of un-updateable TagHelpers.

Reply:
I see your point. Right now this works because we recreate the parser for any type of `ContextChange`. But you are right, we should be able to update TagHelpers without having to create a new parser.

Reply:
Just discussed with @NTaylorMullen. We just want to keep the existing behavior here (recreate the parser whenever there are new TH descriptors). We can revisit this if we see a perf impact in the future.